akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala

```diff
@@ -304,6 +381,26 @@ private[http2] abstract class Http2Demux(http2Settings: Http2CommonSettings, ini
         settingsAppliedOk
       }
 
+      override protected def onTimer(timerKey: Any): Unit = timerKey match {
+        case ConfigurablePing.Tick =>
+          val state = pingState.get // no tick scheduled if state is not set
+
+          // don't do anything unless there are active streams
+          if (activeStreamCount() > 0) {
+            state.onTick()
+            val now = System.nanoTime()
+            if (state.pingAckOverdue(now)) {
+              // FIXME this does not actually close connection currently
+              pushGOAWAY(ErrorCode.CANCEL, "Ping ack timeout")
```

Comment:
`CANCEL` is used to reset streams, not sure if that would match here. Maybe just use `PROTOCOL_ERROR` as a peer is required to ack pings.

Reply:
Still using `CANCEL`?

Reply:
Could have sworn I fixed it already. :/ 

Will fix.